### PR TITLE
Adapt mission control with auv interface

### PR DESCRIPTION
**Pull request by patapufate**
# Description

This PR modifies the mission control behavior to use with auv_interfaces::StateStamped instead of pose_estimator::CorrectedData. The test to be used with this behavior were also updated.

This PR depends on:
- Refactor pose_estimator package #91, #93, #99
- Replace pose_estimator::CorrectedData with auv_interfaces::StateStamped #94

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Linter tests are passing

# How To Test
```sh
rostest mission_control test_mission_control_behaviors.test
```

### CODE REVIEW

File: mission_control/test/test_attitude_servo_behavior.py [hunk 1]
```diff
@@ -53,7 +52,7 @@ class TestAttitudeServoBehavior(unittest.TestCase):
         -   Load the mission attitude_servo_mission_test.xml
         -   Execute the mission
         -   Check if the behavior publishes the goal
-        -   Simulate PoseEstimator data to finish the behavior
+        -   Simulate auv_inteface data to finish the behavior
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:55]: @patapufate nit^N: 
```suggestion
        -   Simulate auv_interfaces/StateStamped data to finish the behavior
```

File: mission_control/test/test_attitude_servo_behavior.py [hunk 2]
```diff
@@ -109,14 +108,11 @@ def attitude_servo_goals_are_set():
...
         # send data to finish the mission
-        rpy_data = Vector3()
-        pose_estimator_corrected_data = CorrectedData()
-        rpy_data.x = 1.0
-        rpy_data.y = 1.0
-        rpy_data.z = 1.0
-        pose_estimator_corrected_data.rpy_ang = rpy_data
-        self.simulated_pose_estimator_pub.publish(
-            pose_estimator_corrected_data)
+        auv_interface_data = StateStamped()
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:111]: @patapufate nit^N:
```suggestion
        msg = StateStamped()
```

**Review by hidmic — APPROVED** [review]
LGTM but for a few minor comments

File: mission_control/test/test_attitude_servo_behavior.py (hunk 1 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:55]: Fixed

File: mission_control/test/test_attitude_servo_behavior.py (hunk 2 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:111]: Fixed

**Pull request merged**